src/main/java/com/es/phoneshop/model/parser/Parser.java

```diff
@@ -0,0 +1,11 @@
+package com.es.phoneshop.model.parser;
+
+
+import com.es.phoneshop.exception.CustomParseException;
+
+import java.util.Locale;
+
+public interface Parser<T> {
+
+    T parse(Locale locale, String str) throws CustomParseException;
```

Comment:
I guess we can leave QuantityParseException or CustomParseException, or?

Reply:
or i think we can delete them both, i made quantity validator return Optional and not to throw any exceptions. I reckoned that if client (our servlets) gets optional empty it is obvious that there was an error parsing value and error message is stored inside map